readme/api/tutorials/cm6_plugin.md

```diff
@@ -0,0 +1,634 @@
+```typescript
+// You may need to add @codemirror/state to package.json
+import { Compartment } from '@codemirror/state';
+
+// ...
+
+	plugin: async (codeMirrorWrapper: any) => {
+		// See https://codemirror.net/examples/config/#compartments
+		const highlightExtension = new Compartment();
+		codeMirrorWrapper.addExtension(highlightExtension.of([]));
+
+		codeMirrorWrapper.defineExtension('myExtension__setHighlightActiveLine', (highlighted: boolean) => {
+			const extension = highlighted ? [ highlightActiveLine() ] : [ ];
+			codeMirrorWrapper.editor.dispatch({
+				effects: [ highlightExtension.reconfigure(extension) ],
+			});
+		});
+	},
+```
+
+In `index.ts`, we could then call the following function [when the plugin's settings change](https://joplinapp.org/api/references/plugin_api/classes/joplinsettings.html#onchange) and after the content script loads:
+```typescript
+const updateContentScriptSettings = async () => {
+	await joplin.commands.execute('editor.execCommand', {
+		name: 'myExtension__setHighlightActiveLine',
+		args: [ await joplin.settings.value(highlightLineSettingId) ],
+	});
+};
+```
+
+</details>
+
+### Style the active line
+
+If you run the plugin, you might notice that the active line has a blue background. Let's customise it with CSS!
+
+There are two different ways of doing this: With a `.css` file and with a [CodeMirror theme](https://codemirror.net/examples/styling/). In this tutorial, we'll use a `.css` file.
+
+Create a new `style.css` file within the `src` directory. Set its content to
+```css
+.cm-editor .cm-line.cm-activeLine {
+	/* See https://joplinapp.org/help/api/references/plugin_theming
+	   for more information about styling with plugins */
+	color: var(--joplin-color);
+	background-color: rgba(200, 200, 0, 0.4);
+}
+```
+
+Next, load the CSS file from the CodeMirror content script:
+```typescript
+import { lineNumbers, highlightActiveLine } from '@codemirror/view';
+
+export default (context: { contentScriptId: string, postMessage: any }) => {
+	return {
+		plugin: async (codeMirrorWrapper: any) => {
+			// ...hidden
+		},
+		assets: () => {
+			return [ { name: './style.css' } ];
+		},
+	};
+};
+```
+
+The active line should now have a light-yellow background, but only when the "highlight active line" setting is enabled.
+
+## CodeMirror 5 compatibility
+
+Joplin's legacy markdown editor uses CodeMirror 5. The beta editor uses CodeMirror 6.
+
+Unfortunately, the [CodeMirror 5 API](https://codemirror.net/5/) and [CodeMirror 6 API](https://codemirror.net/)s are very different. As such, you'll likely need two different content scripts — one for CodeMirror 5 and one for CodeMirror 6. [This pull request](https://github.com/roman-r-m/joplin-plugin-quick-links/pull/15/files#diff-a19ae4175adf4e5173549901c8535f2a45278f8a907da485899660c08c1c520b) provides an example of how CodeMirror 6 support might be added to an existing plugin.
```

Comment:
Maybe clarify here why we have both APIs? As a developer if I was reading this it would make me feel like I don't really want to develop a plugin for this app since supporting two APIs is too much hassle. We also want to encourage developers to use the CM6 API since eventually we only want to support this.

Maybe we should write something like this?

> As of Joplin v2.14 we recommend that you create CM6-based plugins. If you have a requirement to support old Joplin versions however you can target both CM5 and CM6 APIs. Follow the tutorial below for information on how to do this.